pallets/settlement/src/lib.rs

```diff
@@ -760,18 +746,13 @@ decl_module! {
                 <AffirmsReceived>::insert(instruction_id, portfolio, AffirmationStatus::Rejected);
             }
 
-            // Execute the instruction if it was meant to be executed on affirmation
-            let weight_for_instruction_execution = Self::is_instruction_executed(
+            Self::deposit_event(RawEvent::InstructionRejected(did, instruction_id));
+            // Schedule the instruction to execute in the next block only if it was meant to be executed on affirmation.
+            Self::restricted_schedule_instruction(
```

Comment:
There are a few edge cases we need to consider. What if the user does multiple affirmations and withdrawal of affirmations or rejections in the same block? We don't want the instruction to be scheduled multiple times.

Perhaps withdrawal should cancel any pending executions and rejections should ensure that there isn't already a scheduled job.